smdatatools/data_processing/input_processor.py

```diff
@@ -8,7 +8,7 @@ class InputProcessor:
     def convert_note(line: str) -> str:                                                      
         return sub('4', '1', sub('[MKLF]', '0', line))    #replaces extra notes: M, K, L, F; replaces 4 note
 
-    def parse_sm_input(sm_file: list[str]) -> tuple[defaultdict(list), bool]:
+    def parse_sm_input(sm_file: list[str]) -> tuple[dict[str, dict[str or list[str]] or float or int ], bool]:
```

Comment:
Same here

Reply:
Check my commit message. For Python version <3.10, I have to use OR instead of a Union operator. If you can test it from your end with a Python version 3.10+ and it still works then there shouldn't be any problem.